src/Metal/MTLDevice.cs

```diff
@@ -50,13 +50,21 @@ public static partial class MTLDevice {
 				return system_default;
 			}
 		}
+		
+#if MONOMAC || __MACCATALYST__
```

Comment:
This should be run on Mac Catalyst as well now:

https://github.com/xamarin/xamarin-macios/blob/9286bfb79309811b96ece9a283f1d13d5f7c52e3/tests/monotouch-test/Metal/MTLDeviceTests.cs#L24-L38

Reply:
nice catch.